### PR TITLE
Add `DbtDocsGCSOperator`

### PR DESCRIPTION
**Pull request by jbandoro**
## Description

Adds `DbtDocsGCSOperator` so dbt docs can be uploaded to GCS.

## Related Issue(s)

#541 

## Breaking Change?

~~Yes, I decided to create an abstract base class `DbtDocsCloudLocalOperator` since the construction for all of S3, Azure and GCS operators are all so similar, however this breaks `DbtDocsAzureStorageLocalOperator` and `DbtDocsS3LocalOperator` since instead of `aws_conn_id` and `azure_conn_id` it is now `connection_id`.~~

~~Also `DbtDocsAzureStorageLocalOperator` uses `bucket_name` now like the others instead of `container_name`.~~

No breaking changes but standardized `DbtDocsS3LocalOperator`, `DbtDocsAzureStorageLocalOperator` to accept args for `connection_id` and `bucket_name`. The current args of `aws_conn_id` (S3), `azure_conn_id` and `container_name` (Azure) will still work with warnings to switch to `connection_id` and `bucket_name`.

## Checklist

- [x] I have made corresponding changes to the documentation (if required)
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by tatiana — COMMENTED** [review]
Thank you very much @jbandoro; this is looking great!

I particularly enjoyed the refactoring you did. It makes the code easier to maintain.

There are some comments inline, the most important being the backward compatibility.

Please also fix the broken integration test:
```
ERROR tests/test_example_dags_no_connections.py - assert not {'/home/runner/work/astronomer-cosmos/astronomer-cosmos/dev/dags/dbt_docs.py': 'Traceback (most recent call last):\n  ...G")\nairflow.exceptions.DuplicateTaskIdFound: Task id \'generate_dbt_docs_azure\' has already been added to the DAG\n'}
```

File: cosmos/operators/local.py [hunk 1]
```diff
@@ -539,48 +540,54 @@ def __init__(self, **kwargs: Any) -> None:
...
+    @abstractmethod
+    def upload_to_cloud_storage(self, project_dir: str) -> None:
+        pass
+
+
+class DbtDocsS3LocalOperator(DbtDocsCloudLocalOperator):
+    """
+    Executes `dbt docs generate` command and upload to S3 storage. Returns the S3 path to the generated documentation.
+
+    :param connection_id: S3's Airflow connection ID
```
[the remark this answers is not in the document]
Comment (jbandoro) [reply cosmos/operators/local.py:575]: Updated so `aws_conn_id` will still work.

File: cosmos/operators/local.py [hunk 2]
```diff
@@ -599,67 +606,84 @@ def upload_to_s3(self, project_dir: str) -> None:
...
 
-class DbtDocsAzureStorageLocalOperator(DbtDocsLocalOperator):
+class DbtDocsAzureStorageLocalOperator(DbtDocsCloudLocalOperator):
     """
     Executes `dbt docs generate` command and upload to Azure Blob Storage.
 
-    :param azure_conn_id: Azure Blob Storage's Airflow connection ID
-    :param container_name: Azure Blob Storage's bucket name
+    :param connection_id: Azure Blob Storage's Airflow connection ID
+    :param bucket_name: Azure Blob Storage's bucket name
```
[the remark this answers is not in the document]
Comment (jbandoro) [reply cosmos/operators/local.py:634]: Updated so `azure_conn_id` and `container_name` will still work.

File: tests/operators/test_local.py [hunk 3]
```diff
@@ -356,13 +357,16 @@ def test_operator_execute_with_flags(mock_build_and_run_cmd, operator_class, kwa
...
 )
 @patch("cosmos.operators.local.DbtLocalBaseOperator.build_and_run_cmd")
 def test_operator_execute_without_flags(mock_build_and_run_cmd, operator_class):
+    cloud_docs_kwargs = {"connection_id": "fake-conn", "bucket_name": "fake-bucket"}
     operator_class_kwargs = {
-        DbtDocsS3LocalOperator: {"aws_conn_id": "fake-conn", "bucket_name": "fake-bucket"},
-        DbtDocsAzureStorageLocalOperator: {"azure_conn_id": "fake-conn", "container_name": "fake-container"},
+        DbtDocsS3LocalOperator: cloud_docs_kwargs,
+        DbtDocsAzureStorageLocalOperator: cloud_docs_kwargs,
+        DbtDocsGCSLocalOperator: cloud_docs_kwargs,
```
[the remark this answers is not in the document]
Comment (jbandoro) [reply tests/operators/test_local.py]: I added a test for `DbtDocsGCSLocalOperator` that checks that GCSHook is called with the expected connection string and that files are uploaded to the expected paths. Let me know what you think!

**Comment by jbandoro** [comment]
@tatiana the integration tests are failing with: 
```
FAILED tests/test_example_dags.py::test_example_dag[docs_dag] - airflow.exceptions.AirflowException: ('dbt command failed. The command returned a non-zero exit code 2. Details: ', '\x1b[0m06:39:07  Running with dbt=1.6.6', '\x1b[0m06:39:07  Registered adapter: ***=1.6.6', '\x1b[0m06:39:07  Encountered an error:', 'Compilation Error', '  dbt found 1 package(s) specified in packages.yml, but only 0 package(s) installed in dbt_packages. Run "dbt deps" to install package dependencies.')
```
I modified the docs dag but not related to packages. If you could point me to how I should fix this in this PR it would be helpful - thanks!

**Review by tatiana — APPROVED** [review]
Thank you very much, @jbandoro ! Looks great!

**Comment by tatiana** [comment]
@jbandoro, this change is available in the 1.3.0a1 pre-release of Cosmos:
* https://github.com/astronomer/astronomer-cosmos/releases/tag/astronomer-cosmos-v1.3.0a1
* https://pypi.org/project/astronomer-cosmos/1.3.0a1/

Thank you very much for your contribution!

**Pull request merged**